apps/ios/LiveVideo/LiveVideo/Twilio/Sync/SyncManager.swift

```diff
@@ -8,6 +8,7 @@ import TwilioSyncClient
 /// Consolidates configuration and error handling for stores that are backed by [Twilio Sync](https://www.twilio.com/sync).
 class SyncManager: NSObject {
     let errorPublisher = PassthroughSubject<Error, Never>()
+    var isConnected: Bool { client != nil }
```

Comment:
Does the Sync client provide a state that provides a more accurate representation of the connected state?

Reply:
There isn't a simple way to check if the client is not completely dead.

I could probably do:
```swift
var isConnected: Bool {
    guard let client = client else { return false }

    switch client.connectionState {
    case .unknown, .connected, .connecting, .error:
        return true
    case .fatalError, .denied, .disconnected:
        return false
    @unknown default:
        return true
    }
}
```
The `error` state is not fatal but this isn't super clear from API or docs.

And `@unknown default:` is because of Objc bridging. I guess I assume any new state that is added is connected.

And then there is the connection rebalancing that happens hourly that may impact this. It shouldn't go into `disconnected` but sort of hard to test. 

So I tend to think relying on `connectionState` as little as possible is good given everything going on and probably best to just check for nil client. If we receive a fatal error the client is set to nil. 

I could be convinced otherwise though. I understand the intent of making it safer. I'm just not sure the added complexity actually makes it safer.

Reply:
Okay fair enough. I was mainly trying to avoid cases where sync is in a terminal failed state and we proceed with connecting to a room or player.